### PR TITLE
Add Clientset Docs

### PR DESCRIPTION
**Pull request by diyarab**
## What does this PR do?

- [ ] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [x] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- This PR adds documentation for the Chaos Controller clientset. 

## Code Quality Checklist

- [ ] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [ ] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

**Comment by aymericDD** [comment]
Good job 👍 I was expecting a `README.md` inside the `clienset` package. Maybe we can just move the [clientset.md](https://github.com/DataDog/chaos-controller/pull/855/files#diff-70c84b97661d5d63a92b5ff91d0f75047bdf5ec268f55c4ce937e21fee9627be) into the `clienset` package as a `README.md`.

File: docs/clientset.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Clientset
+
+The Chaos Controller clientset is a structured interface that simplifies interactions with the Chaos Controller's Kubernetes resources. It featured dedicated typed clients, which enable straightforward operations on chaos resources such as Disruptions and DisruptionCrons. This document provides further details on the clientset's architecture, its typed clients, and their usage.
+
+## Clientset Generation
```
Comment (aymericDD) [inline docs/clientset.md:5]: In this section we can add a concrete example with the definition of the annotations to defined and also the command line to run WDYT? In the future we will also have to give strait forward method by using the `make` command.
``` bash
client-gen \
--input-base="github.com/DataDog/chaos-controller/api" \
--input="v1beta1" \
--clientset-name="v1beta1" \
--output-dir="$(pwd)/clientset" \
--output-pkg="github.com/DataDog/chaos-controller/clientset" \
--fake-clientset=false
```

File: docs/README.md [hunk 2]
```diff
@@ -4,23 +4,24 @@ This folder contains the detailed documentation of most parts of the project.
...
-  * [Disk Failure](disk_failure.md)
-  * [Disk Pressure](disk_pressure.md)
-  * [DNS Disruption](dns_disruption.md)
-  * [GRPC Disruption](grpc_disruption.md)
-  * [Network Disruption](network_disruption.md)
+- [General Design](design.md)
+- [Configuration Guide](configuration.md)
+- [Features Guide](features.md)
+- [Development Guide](development.md)
+- [Clientset Guide](clientset_usage.md)
```
Comment (ptnapoleon) [inline docs/README.md:11]:
```suggestion
- [Clientset Guide](clientset.md)
```

File: docs/clientset.md (hunk 1 above)
Comment (ptnapoleon) [reply docs/clientset.md:5]: Yes, and even include how to download and install the rc1 version you used to generate these?

File: docs/clientset.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+## Clientset Architecture
+
+A clientset groups together multiple typed clients, each tailored to interact with a distinct set of Kubernetes API resources. For the Chaos Controller, the clientset is designed to encapsulate all the necessary logic for managing custom resources, specifically Disruptions and DisruptionCrons.
+
+These typed clients operate within a specified namespace and offer methods that mirror Kubernetes operations. These operations include `Create`, `Get`, `List`, `Delete`, and `Watch`.
+
+## Usage
+
+Working with Chaos Controller resources via the clientset typically involves the following steps:
```
Comment (ptnapoleon) [inline docs/clientset.md:17]: Could you include code snippets for each of these?

File: docs/clientset.md (hunk 1 above)
Comment (diyarab) [reply docs/clientset.md]: I added a guide for clientset generation. It probably would need to be updated once the Makefile is complete.

**Review by aymericDD — APPROVED** [review]
💯

**Pull request merged**